pkg/controller/cluster/log-search.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2020, MinIO, Inc.
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+
+	miniov1 "github.com/minio/operator/pkg/apis/minio.min.io/v1"
+	"github.com/minio/operator/pkg/resources/services"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// logDB represents tenant's Postgres DB used in Log Search feature
+type logDB struct {
+	connStr string
+}
+
+// newLogDB returns a logDB for a given tenant and secret
+func newLogDB(tenant *miniov1.Tenant, secret *corev1.Secret) *logDB {
+	pgPasswd := string(secret.Data[miniov1.LogPgPassKey])
+	dbAddr := services.GetLogSearchDBAddr(tenant)
+	mainConnStr := fmt.Sprintf("postgres://%s:%s@%s?sslmode=disable", miniov1.LogPgUser, pgPasswd, dbAddr)
+	return &logDB{
+		connStr: mainConnStr,
+	}
+}
+
+// HealthCheck returns nil if tenant's DB is ready and an error otherwise.
+func (ld *logDB) HealthCheck(ctx context.Context) error {
+	conn, err := pgx.Connect(ctx, ld.connStr)
+	if err != nil {
+		return err
+	}
+	conn.Close(ctx) //nolint:errcheck
+	return nil
+}
+
+// CreateAuditDB creates the audit log database in tenant's DB.
+func (ld *logDB) CreateAuditDB(ctx context.Context) error {
```

Comment:
I think this is the only code that needs the `pgx` dependency, if we are using the official postgres image we could set `POSTGRES_DB` and that would create a database with the desired name by default. This way we don't need to add the pgx and libpq dependencies to operator

Reply:
This has been addressed in the recent update to this PR.